src/pages/TrafficList/TrafficListComponent.tsx

```diff
@@ -0,0 +1,335 @@
+  Title,
+  Tooltip,
+  TooltipPosition
+} from '@patternfly/react-core';
+import { style } from 'typestyle';
+import {
+  cellWidth,
+  IRow,
+  ISortBy,
+  sortable,
+  SortByDirection,
+  Table,
+  TableBody,
+  TableHeader
+} from '@patternfly/react-table';
+import { Link } from 'react-router-dom';
+import { TrafficItem, TrafficNode } from './TrafficDetails';
+import * as FilterComponent from '../../components/FilterList/FilterComponent';
+import { ThresholdStatus, NA } from 'types/Health';
+import { NodeType, hasProtocolTraffic, ProtocolTraffic } from 'types/Graph';
+import { getTrafficHealth } from 'types/ErrorRate';
+import history, { URLParam } from 'app/History';
+import { createIcon } from 'components/Health/Helper';
+
+export interface TrafficListItem {
+  healthStatus: ThresholdStatus;
+  icon: string;
+  node: TrafficNode;
+  protocol: string;
+  trafficRate: string;
+  trafficPercent: string; // percent success
+}
+
+type TrafficListComponentProps = FilterComponent.Props<TrafficListItem> & {
+  direction: 'inbound' | 'outbound';
+  trafficItems: TrafficItem[];
+};
+
+type TrafficListComponentState = FilterComponent.State<TrafficListItem> & {
+  sortBy: ISortBy;
+};
+
+const columns = [
+  {
+    title: 'Name',
+    transforms: [sortable]
+  },
+  {
+    title: 'Protocol',
+    transforms: [sortable]
+  },
+  {
+    title: 'Rate',
+    transforms: [sortable]
+  },
+  {
+    title: 'Percent',
+    transforms: [sortable, cellWidth(5) as any]
+  },
+  {
+    title: 'TrafficStatus',
+    transforms: [sortable]
+  },
+  {
+    title: 'Actions'
+  }
+];
+
+// Style constants
+const containerPadding = style({ padding: '20px 20px 20px 20px' });
+
+class TrafficListComponent extends React.Component<TrafficListComponentProps, TrafficListComponentState> {
+  constructor(props: TrafficListComponentProps) {
+    super(props);
+    this.state = {
+      currentSortField: this.props.currentSortField,
+      isSortAscending: this.props.isSortAscending,
+      listItems: [],
+      sortBy: {}
+    };
+  }
+
+  // It invokes backend when component is mounted
+  componentDidMount() {
+    this.updateListItems();
+  }
+
+  componentDidUpdate(prevProps: TrafficListComponentProps, _prevState: TrafficListComponentState, _snapshot: any) {
+    const paramChange =
+      prevProps.trafficItems !== this.props.trafficItems ||
+      prevProps.isSortAscending !== this.props.isSortAscending ||
+      prevProps.currentSortField.title !== this.props.currentSortField.title;
+
+    if (paramChange) {
+      this.setState({
+        currentSortField: this.props.currentSortField,
+        isSortAscending: this.props.isSortAscending
+      });
+
+      this.updateListItems();
+    }
+  }
+
+  render() {
+    return (
+      <div className={containerPadding}>
+        <Table
+          aria-label="Sortable Table"
+          sortBy={this.state.sortBy}
+          cells={columns}
+          rows={this.rows()}
+          onSort={this.onSort}
+        >
```

Comment:
In PF Table component, rows are generated from a structure passed into the component.
For sublist, consider adding the `className="table"` style to fix the vertical align, specially on icons,

Reply:
@lucasponce I modeled this after ExperimentListContainer.  Shall I add the same recommended change in that class as part of this PR?

Reply:
Yes, thanks. I've noticed it has same vertical alignment problem:
![image](https://user-images.githubusercontent.com/1662329/94274978-b2284a80-ff46-11ea-810c-d6cfd2030db2.png)

Reply:
@lucasponce After our off-pr discussion we have decided that the className="table" is not necessary.  So I will ignore.